pydantic_evals/pydantic_evals/dataset.py

```diff
@@ -324,6 +324,8 @@ async def _handle_case(case: Case[InputsT, OutputT, MetadataT], report_case_name
                 trace_id=trace_id,
             )
             if (averages := report.averages()) is not None and averages.assertions is not None:
+                experiment_metadata = {'n_cases': len(self.cases), 'averages': averages}
```

Comment:
From the test below, it looks like we already had `n_cases` in here, do we need to repeat it? Or could we drop it there and have it just under the metadata?